src/python/file-name-conventions.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
+import yaml
+import subprocess
+import re
+import sys
+from pathlib import Path
+
+# Naming conventions
+NAMING_CONVENTIONS = {
+    'camelCase': r'^([a-z]+[a-zA-Z0-9]*)?\.',
+    'snake_case': r'^([a-z]+[a-z0-9_]*)?\.',
+    'kebab-case': r'^([a-z]+[a-z0-9\-]*)?\.',
+    'PascalCase': r'^([A-Z]+[a-zA-Z0-9]*)?\.',
```

Comment:
How about adding the `black` formatter and a Python linter as well?

Reply:
Done! Although there's only really one file to check right now, I added all the previous formatters/linters that were used in the `econia` repo as well as now `autoflake` and `flake8`

Essentially (wrt python), now the `pre-commit` hook runs:

```shell
poetry run python ./src/python/hooks/src/file-name-conventions.py
poetry run autoflake -i --remove-all-unused-imports --remove-unused-variables --ignore-init-module-imports
poetry run black
poetry run flake8
poetry run isort
poetry run mypy
```

on all of the python files passed in from `pre-commit`, which is, as of right now, `--all-files` in the repo

Reply:
Note that the linters/formatters now only run a check in CI, but you can get it to edit things in place with the `src/sh/python-lint/format.sh` script